guillotina/transactions.py

```diff
@@ -131,24 +131,27 @@ async def __aexit__(self, exc_type, exc, tb):
                 self.txn.deleted = {**self.previous_txn.deleted, **self.txn.deleted}
                 self.txn.added = {**self.previous_txn.added, **self.txn.added}
 
-        if self.abort_when_done:
+        if exc:
             await self.tm.abort(txn=self.txn)
```

Comment:
basically i changed the logic to:

if `exc` is set we only abort and restore the context vars. Otherwise, we commit/abort, we clean the previous_txn and we executed futures.

Can you confirm that this is the behavior we want?